dask/blockwise.py

```diff
@@ -1138,16 +1138,14 @@ def _optimize_blockwise(full_graph, keys=()):
                 blockwise_layers.add(dep)
 
                 # traverse further to this child's children
+                output_indices = set(layers[dep].output_indices)
+                input_indices = {
+                    i for _, ind in layers[dep].indices if ind for i in ind
+                }
+                is_io_superset = output_indices.issuperset(input_indices)
                 for d in full_graph.dependencies.get(dep, ()):
                     # Don't allow reductions to proceed
-                    output_indices = set(layers[dep].output_indices)
-                    input_indices = {
-                        i for _, ind in layers[dep].indices if ind for i in ind
-                    }
-
-                    if len(dependents[d]) <= 1 and output_indices.issuperset(
-                        input_indices
-                    ):
+                    if len(dependents[d]) <= 1 and is_io_superset:
```

Comment:
perhaps invert the order?
```suggestion
                    if is_io_superset and len(dependents[d]) <= 1:
```

Reply:
Yes, this would let the interpreter optimize branches out in some cases (if it's capable of that?). Nice.

Reply:
No it can't, but evaluating the bool is faster